src/main/host/process.c

```diff
@@ -150,7 +152,19 @@ static void _process_logReturnCode(Process* proc, gint code) {
                         ((code == 0) ? "success" : "error"), code,
                         process_getName(proc));
 
-        if(code == 0) {
+        gchar* fileName = _process_outputFileName(proc, "exitcode");
+        FILE *exitcodeFile = fopen(fileName, "we");
+        g_free(fileName);
+
+        if (exitcodeFile != NULL) {
+            fprintf(exitcodeFile, "%d", code);
+            fclose(exitcodeFile);
+        } else {
+            warning("Could not open '%s' for writing: %s", mainResultString->str, strerror(errno));
+        }
+
+        // if there was no error or was intentionally killed
+        if(code == 0 || code == 128+SIGKILL) {
```

Comment:
Maybe a TODO here to treat death by SIGKILL as an error (or at least a warning) once we've implemented clean shutdown via SIGTERM?

Reply:
Added.